common/src/main/scala/org/apache/celeborn/common/util/Utils.scala

```diff
@@ -415,7 +415,16 @@ object Utils extends Logging {
   }
 
   def localHostName(conf: CelebornConf): String = customHostname.getOrElse {
-    if (conf.bindPreferIP) localIpAddress.getHostAddress else localIpAddress.getCanonicalHostName
+    if (conf.bindPreferIP) addBracketsIfNeeded(localIpAddress.getHostAddress)
+    else localIpAddress.getCanonicalHostName
+  }
+
+  private[util] def addBracketsIfNeeded(addr: String): String = {
+    if (addr.contains(":") && !addr.contains("[")) {
+      s"[$addr]"
+    } else {
+      addr
+    }
```

Comment:
I actually prefer the previous formulation with regex, it was more robust imo.

Reply:
done